Update documentation for the preCICE website (#39)
precice/micro-manager: README.md

---
title: The Micro Manager
permalink: tooling-micro-manager.html
redirect_from: adapter-openfoam.html
keywords: tooling, micro-manager, two-scale, macro-micro
summary: The Micro Manager is a tool to facilitate solving two-scale (macro-micro) coupled problems using the coupling library preCICE.
---

# The Micro Manager

The Micro Manager is a tool to facilitate solving two-scale (macro-micro) coupled problems using the coupling library [preCICE](https://www.precice.org/).

## Installation

> The Micro Manager is a Python package that can be installed using `pip` or manually. Make sure [preCICE](installation-overview.html) is installed before installing the Micro Manager. The Micro Manager is tested with preCICE version [2.5.0](https://github.com/precice/precice/releases/tag/v2.5.0).

Reviewer: ```suggestion
The Micro Manager is a Python package that can be installed using `pip`. Make sure [preCICE](installation-overview.html) is installed before installing the Micro Manager. The Micro Manager is compatible with preCICE version [2.5.0](https://github.com/precice/precice/releases/tag/v2.5.0).
```

pip install --user micro-manager-precice
```

> Installing preCICE is mandatory, other dependencies will be installed by `pip` if they are not already installed. If you encounter problems in the direct installation, see the [dependencies section](#required-dependencies) below.

Reviewer: ```suggestion
Unless already installed, the dependencies will be installed by `pip` during the installation procedure. preCICE itself needs to be installed separately. If you encounter problems in the direct installation, see the [dependencies section](#required-dependencies) below.
```

Ensure that the following dependencies are installed:

* Python 3 or higher

> * [preCICE](installation-overview.html) Version [2.5.0](https://github.com/precice/precice/releases/tag/v2.5.0)

Reviewer: ```suggestion
* [preCICE](installation-overview.html) [v2.5.0](https://github.com/precice/precice/releases/tag/v2.5.0)
```

* [pyprecice: Python language bindings for preCICE](installation-bindings-python.html)
* [numpy](https://numpy.org/install/)
* [mpi4py](https://mpi4py.readthedocs.io/en/stable/install.html)

#### Build and install the Manager using pip

> After cloning this repository, go to the project directory `micro-manager` and run

Reviewer: ```suggestion
After cloning this repository, go to the directory `micro-manager/` and run
```

```bash
pip install --user .
```

#### Build and install the Manager using Python

> After cloning this repository, go to the project directory `micro-manager` and run

Reviewer: ```suggestion
After cloning this repository, go to the directory `micro-manager/` and run
```

```bash
python setup.py install --user
```

## Using the Micro Manager

The Micro Manager facilitates two-scale coupling between one macro-scale and many micro-scale simulations. It creates instances of several micro simulations and couples them to one macro simulation, using preCICE.

> An existing micro simulation code written in Python needs to be converted into a library with a specific class name and specific function names. The next section describes the required library structure of the micro simulation code. On the other hand, the micro-problem is coupled to preCICE directly. The section [couple your code](couple-your-code-overview.html) of the preCICE documentation gives more details on coupling existing codes.

Reviewer: ```suggestion
An existing micro simulation code needs to be converted into a library with a specific class name which has functions with specific names. The next section describes the required library structure of the micro simulation code. On the other hand, the macro simulation code is coupled to preCICE directly. The section [couple your code](couple-your-code-overview.html) of the preCICE documentation gives more details on coupling existing codes.
```
